testConfig.py

```diff
@@ -262,7 +276,17 @@ def parse(
 
             secondary_network_nad = common.structparse_pop_str(
                 *varg.for_key("secondary_network_nad"),
-                default=f"{namespace}/ocp-secondary",
+                default=None,
+            )
+
+            resource_name = common.structparse_pop_str(
+                *varg.for_key("resource_name"),
+                default=None,
+            )
+
+            namespace = common.structparse_pop_str(
+                *varg.for_key("namespace"),
+                default="default",
```

Comment:
I think you should delete this `structparse_pop_str("namespace")` and delete serializing the namespace in `serialize()`.

Note that `namespace` is already an argument passed in by the caller. While this value comes from YAML, it is defined at a parent/outer/caller level and not here.

These `Conf*` classes mainly aim to represent the content of the YAML. Here, `ConfConnection.namespace` is not actually present in the YAML (at this layer). That is slightly ugly, but OK. Note if you parse the entire YAML, then the value is still fully determined, since the value comes from the YAML (just from another place).

You can come up with schemes how we could also add a namespace to YAML at this place, in particular, you would then here need to set `default=namespace`. But overall, it seems unnecessary. `ConfConnection.namespace` is just a bit odd, and not in YAML (at this layer). It's fine.